recaf-assembler/src/main/java/me/coley/recaf/assemble/suggestions/Suggestions.java

```diff
@@ -0,0 +1,171 @@
+package me.coley.recaf.assemble.suggestions;
+
+import me.coley.recaf.assemble.ast.arch.MethodDefinition;
+import me.coley.recaf.assemble.ast.arch.MethodParameter;
+import me.coley.recaf.assemble.util.ProgramClass;
+import me.coley.recaf.assemble.util.ProgramField;
+import me.coley.recaf.assemble.util.ProgramMethod;
+import me.coley.recaf.util.AccessFlag;
+import me.coley.recaf.util.ClasspathUtil;
+import me.darknet.assembler.instructions.ParseInfo;
+import me.darknet.assembler.parser.Group;
+import me.darknet.assembler.parser.groups.InstructionGroup;
+import org.objectweb.asm.Opcodes;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import static me.coley.recaf.util.ClasspathUtil.Tree;
+
+
+public class Suggestions {
+
+	private final Function<String, ProgramClass> mapper;
+	private final Tree systemClasses = ClasspathUtil.getSystemClasses();
+	private final Tree classes;
+	private MethodDefinition method;
+	public final static TreeSet<String> instructions = new TreeSet<>(ParseInfo.actions.keySet());
+	private final static SuggestionResult EMPTY = new SuggestionResult("", Stream.empty());
+
+	/**
+	 * @param classes
+	 * 		a tree of current user classes
+	 * @param mapper
+	 * 		a function that maps a class name to a {@link ProgramClass} instance
+	 */
+	public Suggestions(Tree classes, Function<String, ProgramClass> mapper, MethodDefinition method) {
+		this.classes = classes;
+		this.mapper = mapper;
+		this.method = method;
+	}
+
+	public void setMethod(MethodDefinition method) {
+		this.method = method;
+	}
+
+	/**
+	 * Supply suggestions for a given group and partial parameters.
+	 *
+	 * @param group
+	 * 		Group to suggest for.
+	 *
+	 * @return Suggestions for the given group.
+	 */
+	public SuggestionResult getSuggestion(Group group) {
+		if(group == null) {
+			return new SuggestionResult("", instructions.stream());
+		}
+		switch (group.type) {
+			case INSTRUCTION:
+				return getInstructionSuggestion(group);
+			case IDENTIFIER: {
+				return new SuggestionResult(
+						group.content(),
+						instructions.stream().filter(s -> s.startsWith(group.content()))
+				);
+			}
+
+		}
+		// return empty suggestion result
+		return new SuggestionResult("", Stream.empty());
+	}
+
+	private SuggestionResult getInstructionSuggestion(Group group) {
+		InstructionGroup instruction = (InstructionGroup) group;
+		Group[] children = instruction.children;
+		String inst = instruction.content();
+		switch (instruction.content()) {
+			case "new":
+			case "anewarray":
+			case "checkcast":
+			case "instanceof":
+				return startsWith(children[0] == null ? "" : children[0].content());
+			case "invokestatic":
+			case "invokeinterface":
+			case "invokespecial":
+			case "invokevirtual": {
```

Comment:
We added a special instruction to JASM to handle ITF=true for non `invokeinterface` instructions, `invokevirtualinterface`. Should add support here too.